lib/ui/src/modules/ui/components/search_box.js

```diff
@@ -62,12 +62,8 @@ const suggestionTemplate = (props, state, styles, clickHandler) =>
   ));
 
 class SearchBox extends React.Component {
-  constructor(props) {
-    super(props);
-
-    const { theme = {} } = this.props;
-
-    this.state = {
+  static getDerivedStateFromProps({ theme }) {
```

Comment:
https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization

Reply:
Note that in React 16.4 `gDSFP` is called as often as `render`, so there's actually no gain in using it like that